audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/KafkaProducerService.java

```diff
@@ -0,0 +1,148 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.audit.service;
+
+import akka.Done;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import uk.gov.gchq.palisade.service.audit.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.audit.model.AuditSuccessMessage;
+import uk.gov.gchq.palisade.service.audit.model.Token;
+import uk.gov.gchq.palisade.service.audit.stream.ConsumerTopicConfiguration;
+import uk.gov.gchq.palisade.service.audit.stream.ProducerTopicConfiguration.Topic;
+
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * A service mimicking the Kafka API to the service.
+ * Write the request and headers to the upstream topic.
+ * These messages will then later be read by the service.
+ * Intended for debugging only.
+ */
+public class KafkaProducerService {
+    private final Sink<ProducerRecord<String, AuditErrorMessage>, CompletionStage<Done>> upstreamErrorSink;
+    private final Sink<ProducerRecord<String, AuditSuccessMessage>, CompletionStage<Done>> upstreamSuccessSink;
+    private final ConsumerTopicConfiguration upstreamConfig;
+    private final Materializer materializer;
+
+    /**
+     * Autowired constructor for the rest controller
+     *
+     * @param upstreamErrorSink   a sink to the upstream error topic
+     * @param upstreamSuccessSink a sink to the upstream success topic
+     * @param upstreamConfig      the config for the topic (name, partitions, ...)
+     * @param materializer        the akka system materializer
+     */
+    public KafkaProducerService(final Sink<ProducerRecord<String, AuditErrorMessage>, CompletionStage<Done>> upstreamErrorSink,
+                                final Sink<ProducerRecord<String, AuditSuccessMessage>, CompletionStage<Done>> upstreamSuccessSink,
+                                final ConsumerTopicConfiguration upstreamConfig,
+                                final Materializer materializer) {
+        this.upstreamErrorSink = upstreamErrorSink;
+        this.upstreamSuccessSink = upstreamSuccessSink;
+        this.upstreamConfig = upstreamConfig;
+        this.materializer = materializer;
+    }
+
+    /**
+     * Takes a list of requests and processes each of them with the given headers.
```

Comment:
mention in the javadoc that this is for Success messages only, same for Error messages in the processErorrRequest

Reply:
Added some extra JavaDoc details